pkg/streamingpromql/functions/math.go

```diff
@@ -57,8 +57,14 @@ var UnaryNegation InstantVectorSeriesFunction = func(seriesData types.InstantVec
 		seriesData.Floats[i].F = -seriesData.Floats[i].F
 	}
 
-	// Prometheus' engine currently leaves histograms as-is for unary negation, so we do the same.
-	// See https://github.com/prometheus/prometheus/pull/14821 for more discussion of this.
+	for i := range seriesData.Histograms {
+		if i > 0 && seriesData.Histograms[i].H == seriesData.Histograms[i-1].H {
+			// Previous point shares the same histogram instance, which we've already negated, so don't negate it again.
```

Comment:
(Not for now, but I'm more and more thinking we should copy histograms for lookbacks. Seems so risky that we'll miss something 😬)

Reply:
Or rather than all lookbacks, perhaps copy histograms if they're used for successive points? Would be interesting to try and see the performance impact.